### ISSUE TITLE
out of range index in a tuple magically works

### ISSUE THREAD

**Issue by aleksey-bykov** [Microsoft/TypeScript#5203]
``` typescript

var x : [number, number] = [1, 2];
var y = x[2]; // expected a compile error, actual ok
```

By [design](https://github.com/aleksey-bykov/by-design-script), isn't it?

**Comment by RyanCavanaugh** [comment]
It is currently per spec, but I don't think this behavior is desirable. This should probably be an error by some new rule.

**Comment by DanielRosenwasser** [comment]
Well the problem is that you're going to get that numeric index signature from `Array<T>`. There is another discussion about having an alternative interface for tuples at https://github.com/Microsoft/TypeScript/issues/4988#issuecomment-145705203.

**Comment by aleksey-bykov** [comment]
I know guys you are trying your best, but I don't see an array, I only see a tuple which is not an array in a broad sense.

**Comment by falsandtru** [comment]
+1 This is invalid behavior.

**Comment by OliverJAsh** [comment]
I ran into exactly the same issue.

**Comment by Igorbek** [comment]
See proposal #6229

**Comment by andy-ms** [comment]
It's now an error for tuples to have extra elements (#17765), so maybe we could revisit this? @sandersn @RyanCavanaugh

**Comment by aleksey-bykov** [comment]
god works mysterious ways
